### ISSUE TITLE
Assembly Alias causes  `System.Text.Json` exception

### ISSUE THREAD

**Issue by bitsandfoxes** [getsentry/sentry-unity#1777]
When building the SDK with TFM `netstandard2.1` (i.e. using Unity 2022 or newer) and then running the aliasing tool causes 
```
TypeLoadException: VTable setup of type System.Text.Json.Utf8JsonWriter failed
```
when the SDK attempts to serialize an envelope.

**Comment by bitsandfoxes** [comment]
As workarounds:
1. Use version Unity 2021 or older to build the SDK - and that's how CI does it
2. Disable aliasing when packing/repacking the SDK

**Comment by sambonfire** [comment]
Workaround #1 won't work for us, so I attempted workaround #2.  It worked in editor, but when I did a player build, it crashes with a new error:

```
System.Text.Json.Utf8JsonWriter doesn't implement interface System.IAsyncDisposable
```

**Comment by bruno-garcia** [comment]
Since we're SOL on this one, I'll try my luck and see if @xoof has any idea what might be going on 🙏 

tl;dr: Same code base, if we build with netstandard2.1 (i.e. using Unity 2022 or newer) it crashes. Works if we build with ns2.0.

**Comment by bitsandfoxes** [comment]
Closing this in favor of https://github.com/getsentry/sentry-unity/issues/2717
